include/rocksdb/advanced_options.h

```diff
@@ -105,9 +105,13 @@ struct CompressionOptions {
   //
   // When compression dictionary is disabled, we compress and write each block
   // before buffering data for the next one. When compression dictionary is
-  // enabled, we buffer all SST file data in-memory so we can sample it, as data
+  // enabled, we buffer SST file data in-memory so we can sample it, as data
   // can only be compressed and written after the dictionary has been finalized.
-  // So users of this feature may see increased memory usage.
+  //
+  // The amount of data buffered can be limited by `max_dict_buffer_bytes`. This
+  // buffered memory is charged to block cache when available. If block cache
```

Comment:
Do you see a need of specifying "when available"? I am thinking something like "..when the block cache is available".

Reply:
Yes, it felt slightly ambiguous to me too. Done, with adjusted phrasing.